api/src/backend/tasks/jobs/report.py

```diff
@@ -0,0 +1,1023 @@
+            spaceBefore=20,
+            spaceAfter=12,
+            fontName="PlusJakartaSans",
+            leftIndent=0,
+            borderWidth=2,
+            borderColor=colors.Color(0.2, 0.4, 0.6),
+            borderPadding=8,
+            backColor=colors.Color(0.95, 0.97, 1.0),
+        )
+
+        h2 = ParagraphStyle(
+            "CustomH2",
+            parent=styles["Heading2"],
+            fontSize=14,
+            textColor=colors.Color(0.3, 0.5, 0.7),
+            spaceBefore=15,
+            spaceAfter=8,
+            fontName="PlusJakartaSans",
+            leftIndent=10,
+            borderWidth=1,
+            borderColor=colors.Color(0.7, 0.8, 0.9),
+            borderPadding=5,
+            backColor=colors.Color(0.98, 0.99, 1.0),
+        )
+
+        h3 = ParagraphStyle(
+            "CustomH3",
+            parent=styles["Heading3"],
+            fontSize=12,
+            textColor=colors.Color(0.4, 0.6, 0.8),
+            spaceBefore=10,
+            spaceAfter=6,
+            fontName="PlusJakartaSans",
+            leftIndent=20,
+        )
+
+        normal = ParagraphStyle(
+            "CustomNormal",
+            parent=styles["Normal"],
+            fontSize=10,
+            textColor=colors.Color(0.2, 0.2, 0.2),
+            spaceBefore=4,
+            spaceAfter=4,
+            leftIndent=30,
+            fontName="PlusJakartaSans",
+        )
+
+        normal_center = ParagraphStyle(
+            "CustomNormalCenter",
+            parent=styles["Normal"],
+            fontSize=10,
+            textColor=colors.Color(0.2, 0.2, 0.2),
+            fontName="PlusJakartaSans",
+        )
+        with rls_transaction(tenant_id, using=READ_REPLICA_ALIAS):
+            provider_obj = Provider.objects.get(id=provider_id)
+            prowler_provider = initialize_prowler_provider(provider_obj)
+            provider_type = provider_obj.provider
+
+            frameworks_bulk = Compliance.get_bulk(provider_type)
+            compliance_obj = frameworks_bulk[compliance_id]
+            compliance_framework = getattr(compliance_obj, "Framework", "N/A")
+            compliance_version = getattr(compliance_obj, "Version", "N/A")
+            compliance_name = getattr(compliance_obj, "Name", "N/A")
+            compliance_description = getattr(compliance_obj, "Description", "")
+
+        logger.info(f"Getting findings for scan {scan_id}")
+        findings_qs = (
+            Finding.all_objects.filter(tenant_id=tenant_id, scan_id=scan_id)
+            .order_by("uid")
+            .iterator()
+        )
+        findings = []
+        with rls_transaction(tenant_id, using=READ_REPLICA_ALIAS):
+            for batch, is_last in batched(findings_qs, DJANGO_FINDINGS_BATCH_SIZE):
+                fos = [
+                    FindingOutput.transform_api_finding(f, prowler_provider)
+                    for f in batch
+                ]
+                findings.extend(fos)
```

Comment:
What happens if we have a scan with 1M findings? This concerns me. What do you think @jfagoagas ?